src/APICore.ts

```diff
@@ -69,14 +69,17 @@ export class APICore {
   }
 
   private get requestHeaders(): Record<string, string> {
+    const {version} = require('../package.json');
+
     const authorizationHeader = {
       Authorization: `Bearer ${this.accessToken}`,
     };
 
     const documentsRequestHeaders = {
       ...authorizationHeader,
-      'Content-Type': 'application/json',
       Accept: 'application/json',
+      'Content-Type': 'application/json',
+      'User-Agent': `CoveoSDKNodeJS/${version}`,
```

Comment:
ain't a fan. Can we append to the default UA instead?

Reply:
This suggestion comes from @pdoyon-coveo and his team. You can review the Google doc here:
https://docs.google.com/document/d/1FM3KDK8jlsGSz2IgtXLYfqz_xjvCFWXsSmHYlVf0rfs/edit

But perhaps there's a discussion to be had before merging/releasing this?

Reply:
I think the Louis-Bompart concern is: was there an existing `User-Agent` field sent by the library? If yes, we should merge them, not replace them with this value. FYI this field support multiple values of `lib/version` if we need it.